### PR TITLE
feat: Add Certified filter to Datasets

### PR DESCRIPTION
**Pull request by hughhhh**
https://user-images.githubusercontent.com/27827808/169415026-f4185481-8bac-44c4-b468-db0c0aa5742f.mov
### SUMMARY

Allowing users to know filter by `Dataset.extra.certification`. Using is like check in the ORM to validate

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by hughhhh** [comment]
@eric-briscoe

**Comment by eschutho** [comment]
This is great! Can you add a test? @AAfghahi wrote some nice tests for his database filters here that you could prob use for inspiration: https://github.com/apache/superset/pull/19051/files#diff-b94e273d2b41d3fe8a552cbf34cfa59ab9495154d16823b411470ae782cea38a

**Comment by eschutho** [comment]
Also I was going to test what happens if the word "certification" is in the warning for example, and if so, we may want to lengthen the search string to "certification:" or even `"certification:"`

**Review by pkdotson — APPROVED** [review]
LGTM!

File: superset/datasets/api.py [hunk 1]
```diff
@@ -195,7 +195,11 @@ class DatasetRestApi(BaseSupersetModelRestApi):
         "owners": RelatedFieldFilter("first_name", FilterRelatedOwners),
         "database": "database_name",
     }
-    search_filters = {"sql": [DatasetIsNullOrEmptyFilter]}
+    search_filters = {
+        "sql": [DatasetIsNullOrEmptyFilter],
+        "id": [DatasetCertifiedFilter],
+    }
+    search_columns = ["id", "database", "owners", "sql", "table_name"]
```
Comment (AAfghahi) [inline superset/datasets/api.py:202]: I believe that this limits search to only these columns, is that the intention?

File: superset/datasets/filters.py [hunk 2]
```diff
@@ -33,3 +35,21 @@ def apply(self, query: Query, value: bool) -> Query:
             filter_clause = not_(filter_clause)
 
         return query.filter(filter_clause)
+
+
+class DatasetCertifiedFilter(BaseFilter):  # pylint: disable=too-few-public-methods
+    name = _("Is certified")
+    arg_name = "dataset_is_certified"
+
+    def apply(self, query: Query, value: Any) -> Query:
```
Comment (AAfghahi) [inline superset/datasets/filters.py:44]: it looks like value here is just a boolean?

File: superset/datasets/api.py (hunk 1 above)
Comment (AAfghahi) [reply superset/datasets/api.py:202]: I did this with my filter PR and it broke a lot of things.

Comment (hughhhh) [reply superset/datasets/api.py:202]: if i don't set the search columns, the new filter won't be referenced. So i'm explicitly setting it here just like the other API classes

**Comment by jinghua-qa** [comment]
LGTM

**Pull request merged**